ext/opcache/jit/zend_jit.c

```diff
@@ -3707,19 +3706,14 @@ void zend_jit_init(void)
 #endif
 }
 
+#if ZEND_VM_KIND != ZEND_VM_KIND_CALL && ZEND_VM_KIND != ZEND_VM_KIND_HYBRID
+# error JIT is compatible only with CALL and HYBRID VM
+#endif
```

Comment:
This could be a configure check, but it's difficult to determine the VM kind in configure. Also, triggering this error is only possible by manually re-generating the VM with `zend_vm_gen.php --with-vm-kind=...`, so it seems fine.